packages/ai-jsx/src/core/opentelemetry.ts

```diff
@@ -31,24 +30,37 @@ interface MemoizedIdAndOwner {
   isOwner: boolean;
 }
 
+// Indicates whether we're currently tracing _any_ component so that we can skip
+// tracing non-root synchronous components.
+const isTracingComponent = createContextKey('Whether the current async context is tracing a component');
+
 // Indicates whether the current async context is responsible for tracing the memoized subtree.
 const activeMemoizedIdStorage = createContextKey('The active MemoizedIdAndOwner');
 
 function spanAttributesForElement(element: Element<any>): opentelemetry.Attributes {
   return { 'ai.jsx.tag': element.tag.name, 'ai.jsx.tree': debug(element, true) };
 }
 
-const getEncoder = _.once(() => getEncoding('cl100k_base'));
-
 export function openTelemetryStreamRenderer(streamRenderer: StreamRenderer): StreamRenderer {
+  const traceEveryComponent = Boolean(getEnvVar('AIJSX_OPENTELEMETRY_TRACE_ALL', false));
```

Comment:
Hmm... this is not a strong objection, but I feel like the more robust thing to do here is use loglevels. But I guess OpenTel and/or DataDog doesn't have a loglevel concept? So even if we were to annotate spans with a loglevel, the consumers we have today wouldn't be able to use that?

Reply:
Yeah I agree -- I looked around for something like that but I don't think that's a thing. I suspect the reason is that spans have a parent-child link, and so if you were to link an INFO span to a DEBUG span, any level based filter would have to buffer non-trivial state in order to patch up the relationships.

So I think any level-ish logic is on our side where we decide to emit them -- if we find ourselves wanting more granularity here we can model this after log levels but right now my philosophy is ~YAGNI.

Reply:
Yep, that's fine with me.